### DISCUSSION TITLE
docker container crash after start.

### DISCUSSION THREAD

**Discussion by Basti77** [dani-garcia/vaultwarden#4195] [Q&A]
I had to update Vaultwarden because the mobile app was no longer connecting to my Docker instance.

The peculiar thing is that I can't reinstall it on my Raspberry Pi 2. The container is created and started, but then it crashes. Unfortunately, I have no idea why this is happening or where to find more information.

Here are the steps on how I create a clean container; the data directory was deleted beforehand.

```
pi@raspberrypi:~ $ docker run -d -it --name bitwarden \
   -e ADMIN_TOKEN=xxx \
   -v /home/pi/vaultwarden_data:/data/ \
   -p 8555:80 \
   -p 3012:3012 \
   --restart=always \
   vaultwarden/server:latest
```
Unable to find image 'vaultwarden/server:latest' locally
latest: Pulling from vaultwarden/server
8d7feeb74478: Pull complete
4552ca59ad6d: Pull complete
fb37d8556b9a: Pull complete
622a2f9eb062: Pull complete
12f58a7c9a51: Pull complete
6b205531682c: Pull complete
Digest: sha256:ab9fe547277245533a28d8e0a0c4a1e1120daf469f983fd683fc13556927d4fe
Status: Downloaded newer image for vaultwarden/server:latest
0c974cedfd1113baae8ca87283c1fe8db9c4f204726afb1ff4ac2825140a2ac7
pi@raspberrypi:~ $ sudo docker ps
CONTAINER ID   IMAGE                       COMMAND       CREATED          STATUS                            PORTS                                                                              NAMES
0c974cedfd11   vaultwarden/server:latest   "/start.sh"   26 seconds ago   Up 2 seconds (health: starting)   0.0.0.0:3012->3012/tcp, :::3012->3012/tcp, 0.0.0.0:8555->80/tcp, :::8555->80/tcp   bitwarden
pi@raspberrypi:~ $ sudo docker ps
CONTAINER ID   IMAGE                       COMMAND       CREATED          STATUS                            PORTS                                                                              NAMES
0c974cedfd11   vaultwarden/server:latest   "/start.sh"   37 seconds ago   Up 2 seconds (health: starting)   0.0.0.0:3012->3012/tcp, :::3012->3012/tcp, 0.0.0.0:8555->80/tcp, :::8555->80/tcp   bitwarden
pi@raspberrypi:~ $ sudo docker ps
CONTAINER ID   IMAGE                       COMMAND       CREATED          STATUS                           PORTS     NAMES
0c974cedfd11   vaultwarden/server:latest   "/start.sh"   43 seconds ago   Restarting (101) 4 seconds ago             bitwarden
pi@raspberrypi:~ $

pi@raspberrypi:~ $ docker info
Client: Docker Engine - Community
 Version:    24.0.7
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.21.0
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 1
 Server Version: 24.0.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dd1e886e55dd695541fdcd67420c2888645a495
 runc version: v1.1.10-0-g18a0cb0
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
 Kernel Version: 5.10.103+
 Operating System: Raspbian GNU/Linux 10 (buster)
 OSType: linux
 Architecture: armv6l
 CPUs: 1
 Total Memory: 430MiB
 Name: raspberrypi
 ID: 31892fb0-f1fd-4760-bec1-fe1b13f08219
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

**Comment by BlackDex** [comment]
Try to run without `-d` and see what the reason is it stops.

**Reply by BlackDex** [reply]
Also, i quickly tested it locally on my rpi2, works just fine :)

**Comment by Basti77** [comment]
I have tried it with another 100% free port but always get the same strange message: 

pi@raspberrypi:~ $ sudo docker run --name vaultwarden -v /vw-data/:/data/ --restart unless-stopped -p 82:80 vaultwarden/server:latest
thread 'main' panicked at library/std/src/sys/unix/time.rs:414:76:
called `Result::unwrap()` on an `Err` value: Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**Reply by BlackDex** [reply]
Try the `-alpine` tagged images.
But either your docker or your host is too old, try to update them of possible.

**Comment by Basti77** [comment]
I found this and am trying an update on the bull's eye

https://github.com/pi-hole/docker-pi-hole/releases/tag/2022.04.2

**Comment by Basti77** [comment]
After 3 hours of updating on the old thing, VW is finally running again.
